api-tests/core/content-manager/find-relations.test.api.js

```diff
@@ -0,0 +1,803 @@
+        });
+
+        expect(res.status).toBe(400);
+        expect(res.body).toMatchObject({
+          data: null,
+          error: {
+            details: {},
+            message: `The relational field unknown doesn't exist on ${compoUid}`,
+            name: 'ValidationError',
+            status: 400,
+          },
+        });
+      });
+
+      test('Fail when the field exists but is not a relational field', async () => {
+        const res = await rq({
+          method: 'GET',
+          url: `/content-manager/relations/${compoUid}/${data.shops.draft[0].myCompo.id}/name`,
+        });
+
+        expect(res.status).toBe(400);
+        expect(res.body).toMatchObject({
+          data: null,
+          error: {
+            details: {},
+            message: `The relational field name doesn't exist on ${compoUid}`,
+            name: 'ValidationError',
+            status: 400,
+          },
+        });
+      });
+    });
+  });
+
+  // Run tests against every type of relation in the shop content type, always
+  // from the default locale
+  const defaultLocale = 'en';
+  describe.only.each([
+    ['products_ow', false],
+    ['products_oo', false],
+    ['products_mo', false],
+    ['products_om', false],
+    ['products_mm', false],
+    ['products_mw', false],
+    ['compo_products_ow', true],
+    ['compo_products_mw', true],
+  ])('Relational field (%s) (is in component: %s)', (fieldName, isComponent) => {
+    // Perform the same tests for both draft and published entries
+    // Components don't have a published status
+    const statuses = isComponent ? [['draft']] : [['draft'], ['published']];
+
+    describe.each(statuses)(`Get %s relation(s)`, (status) => {
+      const qs = {
+        status,
+        locale: defaultLocale,
+      };
+
+      describe('Find Available', () => {
+        describe.each([[true], [false]])(
+          'Can retrieve all available relation(s)',
+          (useEmptyShop) => {
+            test(`when entity ID is ${
+              !useEmptyShop ? 'undefined' : 'an empty entity'
+            }`, async () => {
+              const { modelUID, idEmptyShop } = isComponent
+                ? data.testData.component
+                : data.testData.entity;
+              const id = useEmptyShop ? idEmptyShop : undefined;
+
+              const res = await rq({
+                method: 'GET',
+                url: `/content-manager/relations/${modelUID}/${fieldName}`,
+                qs: {
+                  ...qs,
+                  id,
+                },
+              });
+              expect(res.status).toBe(200);
+
+              const productsInThisLocale = data.products[status].filter(
+                // This test is running for the default locale (en)
+                // any products that are in the non default locale should not
+                // be considered available
+                (product) => product.locale === defaultLocale
+              );
```

Comment:
shouldn't this endpoint not return any relation thats not targeting the provided locale?

Reply:
ah! missed data.products is where you store all of the created products :D